airflow/contrib/hooks/gcp_transfer_hook.py

```diff
@@ -56,52 +85,282 @@ def get_conn(self):
                                http=http_authorized, cache_discovery=False)
         return self._conn
 
-    def create_transfer_job(self, description, schedule, transfer_spec, project_id=None):
-        transfer_job = {
-            'status': 'ENABLED',
-            'projectId': project_id or self.project_id,
-            'description': description,
-            'transferSpec': transfer_spec,
-            'schedule': schedule or self._schedule_once_now(),
-        }
-        return self.get_conn().transferJobs().create(body=transfer_job).execute()
-
-    def wait_for_transfer_job(self, job):
+    @GoogleCloudBaseHook.convert_http_exception
+    def create_transfer_job(self, body):
+        """
+        Creates a transfer job that runs periodically.
+
+        :param body: A request body, as described in
+            https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferJobs/patch#request-body
+        :type body: dict
+        :return: If successful, TransferJob.
+        :rtype: dict
+        """
+        body = self._append_project_id(body, 'body project_id parameter')
+        return self.get_conn()\
+            .transferJobs()\
+            .create(body=body)\
+            .execute(num_retries=NUM_RETRIES)
```

Comment:
I personally like when liquid API is always done this way. This is much clearer for me. The officia code style for Python -  PEP8 says  that the line length is 80 characters. In this case, a line break will always occur.

Writing in one line is problematic for version control systems.  If we have a lot of code on one line, there are more changes in it. In addition, unnecessary changes appear - changes that add new lines to meet line length requirements. If you add a new line before, this problem does not exist.

It is worth citing here the recommendations of PEP20, which does not describe the style of the code, but the philosophy behind Python. So these are not strict rules, but they help to understand the environment better.
> Sparse is better than dense.~ PEP 20 - The Zen of Python 
I understand that it is good practice to add new lines if you think that this will improve the legibility of the code. 

FYI: You can add brackets around the query, then you do not need to add a slash.
Something similar to:
````
        return (self.get_conn()
            .transferJobs()
            .create(body=body)
            .execute(num_retries=NUM_RETRIES))
````

Reply:
Hmm, yeah I can see what you mean with version control and line breaks. I just hate those pesky backslashes! 😂 

But I quite like this solution with brackets. Maybe we could use this?

Reply:
The argument about personal visual perceptions does not appeal to me. These are individual matters and if we follow them we will never come to a conclusion. I try to refer to other documents or materials to maintain objectivity. On the other hand, the code is made up of people, and so it must be readable to them. However, everyone can not be satisfied, so referring to the recommendations of other big companies is a good idea.

However, I still thought about it and analyzed how it is done in other places in the project's code.
Look at: https://github.com/apache/airflow/search?q=session.query&unscoped_q=session.query
In particular, for these cases:
https://github.com/apache/airflow/blob/491fd743da36b41b7782a371193bbafa84ef5251/scripts/perf/scheduler_ops_metrics.py#L157-L162
https://github.com/apache/airflow/blob/491fd743da36b41b7782a371193bbafa84ef5251/scripts/perf/scheduler_ops_metrics.py#L71-L76
https://github.com/apache/airflow/blob/491fd743da36b41b7782a371193bbafa84ef5251/scripts/perf/scheduler_ops_metrics.py#L111-L117
https://github.com/apache/airflow/blob/6218d91df503821965c743992f4929900972f037/airflow/www/views.py#L110-L118

My result: I will add parentheses.

Reply:
Ok. Nice research job there.